src/engine/SpatialJoinAlgorithms.cpp

```diff
@@ -223,3 +224,259 @@ Result SpatialJoinAlgorithms::S2geometryAlgorithm() {
+  // maxDist and a safety margine from that
+  double antiDist =
+      (circumferenceMin_ / 2.0) - static_cast<double>(maxDist.value()) * 1.01;
+  // use the bigger circumference as an additional safety margin, use 2.01
+  // instead of 2.0 because of rounding inaccuracies in floating point
+  // operations
+  double distToAntiPoint = (360 / circumferenceMax_) * (antiDist / 2.01);
+  double upperBound = antiPoint.get<1>() + distToAntiPoint;
+  double lowerBound = antiPoint.get<1>() - distToAntiPoint;
+  double leftBound = antiPoint.get<0>() - distToAntiPoint;
+  double rightBound = antiPoint.get<0>() + distToAntiPoint;
+  bool northPoleTouched = false;
+  bool southPoleTouched = false;
+  bool boxCrosses180Longitude = false;  // if the 180 to -180 line is touched
+  // if a pole is crossed, ignore the part after the crossing
+  if (upperBound > 90) {
+    upperBound = 90;
+    northPoleTouched = true;
+  }
+  if (lowerBound < -90) {
+    lowerBound = -90;
+    southPoleTouched = true;
+  }
+  if (leftBound < -180) {
+    leftBound += 360;
+  }
+  if (rightBound > 180) {
+    rightBound -= 360;
+  }
+  if (rightBound < leftBound) {
+    boxCrosses180Longitude = true;
+  }
+  // compute bounding boxes using the anti bounding box from above
+  std::vector<box> boxes;
+  if (!northPoleTouched) {
+    // add upper bounding box(es)
+    if (boxCrosses180Longitude) {
+      boxes.emplace_back(point(leftBound, upperBound), point(180, 90));
+      boxes.emplace_back(point(-180, upperBound), point(rightBound, 90));
+    } else {
+      boxes.emplace_back(point(leftBound, upperBound), point(rightBound, 90));
+    }
+  }
+  if (!southPoleTouched) {
+    // add lower bounding box(es)
+    if (boxCrosses180Longitude) {
+      boxes.emplace_back(point(leftBound, -90), point(180, lowerBound));
+      boxes.emplace_back(point(-180, -90), point(rightBound, lowerBound));
+    } else {
+      boxes.emplace_back(point(leftBound, -90), point(rightBound, lowerBound));
+    }
+  }
+  // add the box(es) inbetween the longitude lines
+  if (boxCrosses180Longitude) {
+    // only one box needed to cover the longitudes
+    boxes.emplace_back(point(rightBound, -90), point(leftBound, 90));
+  } else {
+    // two boxes needed, one left and one right of the anti bounding box
+    boxes.emplace_back(point(-180, -90), point(leftBound, 90));
+    boxes.emplace_back(point(rightBound, -90), point(180, 90));
+  }
+  return boxes;
+}
+
+// ____________________________________________________________________________
+bool SpatialJoinAlgorithms::containedInBoundingBoxes(
+    const std::vector<box>& bbox, point point1) const {
+  // correct lon bounds if necessary
+  while (point1.get<0>() < -180) {
+    point1.set<0>(point1.get<0>() + 360);
+  }
+  while (point1.get<0>() > 180) {
+    point1.set<0>(point1.get<0>() - 360);
+  }
+  if (point1.get<1>() < -90) {
+    point1.set<1>(-90);
+  } else if (point1.get<1>() > 90) {
+    point1.set<1>(90);
+  }
```

Comment:
You have these corrections also elsewhere, maybe you can extract a function for this.

Reply:
i don't think that's possible, because this is the only general implementation. The other cases require extra flags, which get set, when a certain condition happens (i.e. the flag 'northPoleTouched'). I could implement the function in a way, that it returns the flags as well, but then all places would need dummyVariables, which accept the other flags. I think in total it would make the code less readable